tests/integration/testecbug.nim

```diff
@@ -33,8 +33,9 @@ marketplacesuite "Bug #821 - node crashes during erasure coding":
     let cid = clientApi.upload(data).get
 
     var requestId = none RequestId
-    proc onStorageRequested(event: StorageRequested) {.raises:[].} =
-      requestId = event.requestId.some
+    proc onStorageRequested(eventRes: ?!StorageRequested)=
+      if event =? eventRes:
+        requestId = event.requestId.some
```

Comment:
Wdyt about raising a Defect (and fail the test) if there's an error in the callback?

```suggestion
    proc onStorageRequested(event: ?!StorageRequested)=
      requestId = some event.get.requestId
```